### DISCUSSION TITLE
Tooltip explanations for r9k options

### DISCUSSION THREAD

**Discussion by azizLIGHT** [Chatterino/chatterino2#5284] [Ideas] [answered]
It is unclear what changing r9k options does to similar messages. Some description about what each option does could be displayed in a mouseover tooltip like the other options on the settings dialog. 
* `By the same user`: does it mean that if checked the r9k hiding will only apply on a per-user basis? That it would hide the same message by the same user if seen again. And does it mean that if unchecked, that the same message from multiple users will be considered similar and hidden for all of them. Sample tooltip: If enabled, similar messages are hidden on on a per-user basis. If a specific user posts another message that meets the similarity threshold, that message will be hidden. If not enabled, similar messages are hidden across all users. If any user posts a message that met the similarity threshold of a previous message from any user, this message will be hidden.
* `Similarity threshold`: does 0.9 mean that the content of the message compared to the last message is at least 90% the same, character by character, for it to be considered similar and hidden? That it repeated the same emote just multiple times? That the message is the same message, but copy pasted multiple times? How is similiarity threshold determined. 
* `Maximum delay between messages`: 5s to 120s. Is this about how much time a similarity comparison should consider? That a comparison of every new message should be compared to newer messages for 5 seconds at most? So that would be a short comparison timeframe, and that a new comparison for a new message lasts for another 5 seconds? If 120s is selected, what exactly is happening
* `Amount of previous messages to check`: Does this setting get modified by `By the same user`?

**Comment by pajlada** [comment, marked as answer]
Should be improved by https://github.com/Chatterino/chatterino2/pull/5286

> Amount of previous messages to check: Does this setting get modified by By the same user?

If "By the same user" is checked, it's user-based yes
